packages/core/src/submodules/httpAuthSchemes/aws_sdk/resolveAwsSdkSigV4Config.ts

```diff
@@ -106,29 +106,33 @@ export const resolveAwsSdkSigV4Config = <T>(
 ): T & AwsSdkSigV4AuthResolvedConfig => {
   let isUserSupplied = false;
   // Normalize credentials
-  let normalizedCreds: AwsCredentialIdentityProvider | undefined;
+  let credentialsProvider: AwsCredentialIdentityProvider | undefined;
   if (config.credentials) {
     isUserSupplied = true;
-    normalizedCreds = memoizeIdentityProvider(config.credentials, isIdentityExpired, doesIdentityRequireRefresh);
+    credentialsProvider = memoizeIdentityProvider(config.credentials, isIdentityExpired, doesIdentityRequireRefresh);
   }
-  if (!normalizedCreds) {
+  if (!credentialsProvider) {
     // credentialDefaultProvider should always be populated, but in case
     // it isn't, set a default identity provider that throws an error
     if (config.credentialDefaultProvider) {
-      normalizedCreds = normalizeProvider(
+      credentialsProvider = normalizeProvider(
         config.credentialDefaultProvider(
           Object.assign({}, config as any, {
             parentClientConfig: config,
           })
         )
       );
     } else {
-      normalizedCreds = async () => {
+      credentialsProvider = async () => {
         throw new Error("`credentials` is missing");
       };
     }
   }
 
+  const boundCredentialsProvider = async () => {
+    return credentialsProvider!({ callerClientConfig: config });
+  };
```

Comment:
nit: brackets can be removed
```suggestion
  const boundCredentialsProvider = async () => credentialsProvider!({ callerClientConfig: config });
```